src/sedtrails/particle_tracer/data_retriever.py

```diff
@@ -1,5 +1,164 @@
 """
 Flow Field Data Retriever
-=========================
-Retrieve flow field data from the transport converter or the simulation caching and state tracker.
-"""
\ No newline at end of file
+
+Retrieves flow field data from the transport converter or the simulation caching and state tracker.
+Performs temporal interpolation to provide accurate flow field data at any time point for particle tracing.
+"""
+from typing import Tuple
+import numpy as np
+import sys
+from pathlib import Path
+
+# Add parent directory to path to allow importing from other packages
+sys.path.append(str(Path(__file__).parent.parent.parent))
+from sedtrails.transport_converter.format_converter import SedtrailsData
+
+
+class FlowFieldDataRetriever:
+    """
+    Retrieves and interpolates flow field data from SedtrailsData.
+    
+    This class provides methods to extract flow fields at specific times,
+    performing temporal interpolation as needed. Currently supports depth-averaged
+    flow velocity.
+    """
+    
+    def __init__(self, sedtrails_data: SedtrailsData):
+        """
+        Initialize the FlowFieldDataRetriever.
+        
+        Parameters:
+        -----------
+        sedtrails_data : SedtrailsData
+            The SedtrailsData object containing the flow fields
+        """
+        self.sedtrails_data = sedtrails_data
+        
+        # Only using depth-averaged flow velocity for now
+        self.flow_field_name = "depth_avg_flow_velocity"
+    
+    def get_interpolation_indices(self, target_time: float) -> Tuple[int, int, float]:
+        """
+        Get indices for interpolation between two time steps.
+        
+        Parameters:
+        -----------
+        target_time : float
+            Target time in seconds since reference_date
+            
+        Returns:
+        --------
+        Tuple[int, int, float]
+            (lower_index, upper_index, weight) where weight is the interpolation factor [0-1]
+        """
+        times = self.sedtrails_data.times
+        
+        # Handle edge cases
+        if target_time <= times[0]:
+            return 0, 0, 0.0
+        
+        if target_time >= times[-1]:
+            last_idx = len(times) - 1
+            return last_idx, last_idx, 1.0
```

Comment:
The 1.0 here is a 'magic number'. Magic numbers are constants that have no context (we do know know what they represent,  or why is not 2.0 instead of 1.0), and their use should be avoided. A way to avoid them is to use symbolic constants. See the Python example here: https://refactoring.guru/replace-magic-number-with-symbolic-constant